docs/semantics.md

```diff
@@ -248,6 +248,16 @@ However, implicit directories does have drawbacks:
 
 Alternatively, users can create a script which lists the buckets and creates the appropriate objects for the directories so that the ```--implicit-dirs``` flag is not used.
 
+**Using Cloud Storage Buckets with Hierarchical Namespaces Enabled :**
+
+Cloud Storage Fuse also offers seamless support for buckets with hierarchical namespaces enabled. Mounting an HNS-enabled bucket using gcsfuse works exactly like mounting a standard bucket, with no additional configuration required.
+
+HNS-enabled buckets offer several advantages over standard buckets when used with cloud storage fuse:
+
+- HNS-enabled buckets eliminates the need for --implicit-dirs flag. HNS buckets inherently understand directories, so gcsfuse no longer needs this flag to simulate directory structures. Users will see consistent directory listings with or without the flag.
+- In HNS buckets, renaming a folder and its child folders is an atomic operation, meaning all associated resources—including objects and managed folders—are renamed in a single step. This ensures data consistency and significantly improves operation performance.
+- HNS buckets treat folders as first-class entities, closely aligning with traditional file system semantics. Commands like mkdir now directly create folder resources within the bucket, unlike with traditional buckets where directories were simulated using prefixes and 0-byte objects.
+- The [Objects.list](https://cloud.google.com/storage/docs/json_api/v1/objects/list) api is replaced with more efficient [Folder:get](https://cloud.google.com/storage/docs/json_api/v1/folders/getfoldermetadata) api during lookup operations.
```

Comment:
I think we can leave out list Objects. Its a little bit of internal detail.
We can just mention we use GetFolderMetadata which is more efficient and inline to fetch metadata for lookup operations.

Reply:
Lets just use what we did in cloud docs:
[List object](https://pkg.go.dev/cloud.google.com/go/storage#hdr-Listing_objects) calls (BucketHandle.Objects), are converted to [get folder](https://pkg.go.dev/cloud.google.com/go/storage@v1.43.0/control/apiv2#StorageControlClient.GetFolder) calls, resulting in quicker response times and fewer overall list calls for every operation.

Reply:
Adding same as cloud with minor tweakes, please see @marcoa6 , I felt converted is a bit ambiguous so used replaced and specified every operation as every lookup operation to be precise.

[List object](https://pkg.go.dev/cloud.google.com/go/storage#hdr-Listing_objects) calls (BucketHandle.Objects), are converted to [get folder](https://pkg.go.dev/cloud.google.com/go/storage@v1.43.0/control/apiv2#StorageControlClient.GetFolder) calls, resulting in quicker response times and fewer overall list calls for every lookup operation.